### PR TITLE
Auto docker image

### PR DESCRIPTION
**Pull request by bdevans**
This automatically builds a `chaste/develop` image after each PR merge / push which is tagged with the sha hash of the commit.

### CODE REVIEW

File: .github/workflows/docker-images.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Docker Multi-Arch Develop Image CI
+
+on:
+  push:
+    branches: [ "main", "develop" ]
```
Comment (fcooper8472) [inline .github/workflows/docker-images.yml]: Should this just be develop?

File: .github/workflows/docker-images.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    # tags: "v*"
```
Comment (fcooper8472) [inline .github/workflows/docker-images.yml]: Remove?

**Comment by bdevans** [comment]
One of the coverage tests failed due to lack of disk space:
```
376/380 Test #534: TestDiskSpace .....................................................***Failed    0.07 sec
Entering Test that there is plenty of disk available for Chaste.
Free space on Chaste source partition: 4.813652038574219GB
Traceback (most recent call last):
  File "/__w/Chaste/Chaste/python/test/TestDiskSpace.py", line 55, in TestDiskSpace
    self.failIf(source_free_gb < 5,
AssertionError: True is not false : The disk containing the Chaste source tree has less than 5GB of space left.
Failed
```
@jmpf Has `lofty` got enough free disk space and has enough been allocated to Docker?

**Comment by bdevans** [comment]
Also one of the `Continuous` tests:
```
The following tests FAILED:
	 77 - TestNonCachedTetrahedralMesh (Failed)
```

**Comment by jmpf** [comment]
> @jmpf Has `lofty` got enough free disk space and has enough been allocated to Docker?

Lofty has two partitions with over 100 Gb each.  I don't know how docker has been set up by @fcooper8472 or how to reconfigure it though.  The test only just fails so I assume there a configure file which needs boosting by 1 or 2 Gb.

**Comment by fcooper8472** [comment]
Strange...

```
/dev/sda2       215G   92G  112G  46% /home
```

so there's LOADS of free space. Very strange... I'll have a poke around.

**Review by fcooper8472 — APPROVED** [review]

**Pull request merged**